File: testapp/platforms/ios/TestApp/Plugins/com-optimobilecreate-plugins-swift/OptimobileSwift.swift
import OptimoveSDK
@objc(Optimobile_Cordova) class OptimobileSwift : CDVPlugin {
    @objc(initialize:)
    func initialize(command: CDVInvokedUrlCommand) {
        let args = command.arguments
        let config = OptimoveConfigBuilder(optimoveCredentials: args?[0] as? String, optimobileCredentials: args?[1] as? String).enableInAppMessaging(inAppConsentStrategy: .explicitByUser).build()
        Optimove.initialize(with: config)
    }
    
    @objc(setInstallId:)
    func setInstallId(command: CDVInvokedUrlCommand) {
        Optimove.shared.setUserId(command.arguments[0] as? String ?? "")
    }
    
    @objc(pushCampaignsIsEnabled:)
    func pushCampaignsIsEnabled(command: CDVInvokedUrlCommand) {
        OptimoveInApp.updateConsent(forUser: command.arguments[0] as? Bool ?? false)
    }
    
    @objc(reportEvent:)
    func reportEvent(command: CDVInvokedUrlCommand) {
        let name = command.arguments[0] as! String
        var parms = [String: Any]()
        for i in 1..<command.arguments.count {
            parms[command.arguments[i] as! String] = command.arguments[i + 1]
        }
        Optimove.shared.reportEvent(name: name, parameters: parms)
    }
    
    @objc(reportScreenVisit:)
    func reportScreenVisit(command: CDVInvokedUrlCommand) {
        let screenTitle = command.arguments.first!
        let screenCategory = command.arguments[1]
        Optimove.shared.reportScreenVisit(screenTitle: screenTitle as! String, screenCategory: screenCategory as? String)
    }
    
    @objc(registerUser:)
    func registerUser(command: CDVInvokedUrlCommand) {
        let sdkId = command.arguments.first!
        let email = command.arguments[1]
        Optimove.shared.registerUser(sdkId: sdkId as! String, email: email as! String)
    }
    
    @objc(setUserId:)
    func setUserId(command: CDVInvokedUrlCommand) {
        Optimove.shared.setUserId(command.arguments.first as! String)
    }
    
    @objc(getVisitorID)
    func getVisitorID() -> String? {
        return Optimove.getVisitorID()
    }
    
    @objc(setUserEmail:)
    func setUserEmail(command: CDVInvokedUrlCommand) {
        Optimove.shared.setUserEmail(email: command.arguments.first as! String)
    }
    
//    @objc(getInboxItems:)
//    func getInboxItems(command: CDVInvokedUrlCommand) -> [InAppInboxItem] {
//       return OptimoveInApp.getInboxItems()
//    }
}

File: testapp/platforms/ios/www/plugins/com-optimobilecreate-plugins-swift/www/OptimobileSwift.js

cordova.define("com-optimobilecreate-plugins-swift.OptimobileSwift", function(require, exports, module) {
    var exec = require('cordova/exec');
    
    exports.setInstallId = function (arg0, success, error) {
        exec(success, error, 'OptimobileSwift', 'setInstallId', [arg0]);
    };
    
    exports.initialize = function (arg0, arg1, success, error) {
        exec(success, error, 'OptimobileSwift', 'initialize', [arg0, arg1]);
    };
    
    exports.updateConsent = function (arg0, success, error) {
        exec(success, error, 'OptimobileSwift', 'updateConsent', [arg0]);
    };
    
    exports.reportEvent = function (arg0, success, error) {
        exec(success, error, 'OptimobileSwift', 'reportEvent', [arg0]);
    };
    
    exports.reportScreenVisit = function (arg0, success, error) {
        exec(success, error, 'OptimobileSwift', 'reportScreenVisit', [arg0]);
    };
    
    exports.setUserId = function (arg0, success, error) {
        exec(success, error, 'OptimobileSwift', 'setUserId', [arg0]);
    };
    
    exports.registerUser = function (arg0, success, error) {
        exec(success, error, 'OptimobileSwift', 'registerUser', [arg0]);
    };
    
    exports.getVisitorID = function() {
        return new Promise(function (resolve, reject){
            exec(resolve, reject, 'OptimobileSwift', 'getVisitorID');
        })
    };
    
    exports.setUserEmail = function (arg0, success, error) {
        exec(success, error, 'OptimobileSwift', 'setUserEmail', [arg0]);
    };
    
    exports.getInboxItems = function (arg0, success, error) {
        exec(success, error, 'OptimobileSwift', 'getInboxItems', [arg0]);
    };
});

